File: fixtures/scale/ruby/large/lib/check_byteflip.rb
require_relative 'byteflip_encode'
require_relative 'byteflip_decode'
require_relative 'corpus_codec'

def check_byteflip
  corpus_codec.each do |text|
    packed = byteflip_encode(text)
    return false if byteflip_decode(packed) != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_byteflip.py
from cipherlab.ciphers.byteflip import byteflip_encode, byteflip_decode
from cipherlab.specs.byteflip_spec import byteflip_spec
from cipherlab.corpus.corpus_codec import corpus_codec


def check_byteflip():
    spec = byteflip_spec()
    for text in corpus_codec():
        packed = byteflip_encode(text)
        if byteflip_decode(packed) != text:
            return False
    return spec["category"] == "codec"
